data_loader.py

```diff
@@ -0,0 +1,246 @@
+                .agg(['max', 'mean'])
+                .rename(columns={'max': 'insitu_max', 
+                                 'mean': 'insitu_mean'}))
+        return self._insitu_height_stats
+    
+    @property
+    def height_stats(self):
+        """
+        Calculate max and mean tree height from lidar and insitu data
+    
+        Parameters
+        ----------
+        lidar_chm_stats: property
+            Property that creates gdf of LiDAR max and mean tree height
+
+        insitu_height_stats: property
+            Property that creates df of insitu max and mean tree height
+
+        Returns
+        ---------
+        self._height_stats: gdf
+            geodataframe containing plot IDs and lidar and insitu
+            max and mean canopy height
+        """
+        if self._height_stats is None:
+            self._height_stats = (
+                self.lidar_chm_stats
+                .merge(self.insitu_height_stats,
+                       right_index=True,
+                       left_on=self.id_col_name))
+        return self._height_stats
+    
+    def plot_height(self, site_name, ax1max, ax2max):
+        """Plots the insitu vs lidar max and mean canopy height
+
+        Parameters
+        ----------
+        site_name: str
+            name of the site
+
+        ax1max: int 
+            maximum lim for max height data 
+
+        ax2max: int
+            maximum lim for mean height data 
+
+        """
+
+        fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(12,4))
+
+        ax1.scatter(self.height_stats.lidar_max, self.height_stats.insitu_max)
+        ax1.plot((0,1), (0,1), transform=ax1.transAxes, ls='--', c='k')
+        sns.regplot(x='lidar_max', y='insitu_max',
+                    data=self.height_stats,
+                    color='blue',
+                    ax=ax1)
+        ax1.set(xlim=(0,ax1max), ylim=(0,ax1max), aspect='equal',
+               title=('Max canopy height: lidar vs insitu at {}'
+                      .format(site_name)),
+               xlabel='Lidar max height (m)',
+               ylabel='In-situ max height (m)')
+        ax1.xaxis.label.set_fontsize(14)
+        ax1.yaxis.label.set_fontsize(14)
+        ax1.title.set_fontsize(14)
+        
+        ax2.scatter(self.height_stats.lidar_mean, self.height_stats.insitu_mean)
+        ax2.plot((0,1), (0,1), transform=ax2.transAxes, ls='--', c='k')
+        sns.regplot(x='lidar_mean', y='insitu_mean',
+                    data=self.height_stats,
+                    color='blue',
+                    ax=ax2)
+        ax2.set(xlim=(0,ax2max), ylim=(0,ax2max), aspect='equal',
+               title=('Mean canopy height: lidar vs insitu at {}'
+                      .format(site_name)),
+               xlabel='Lidar mean height (m)',
+               ylabel='In-situ mean height (m)')
+        ax2.xaxis.label.set_fontsize(14)
+        ax2.yaxis.label.set_fontsize(14)
+        ax2.title.set_fontsize(14)
+
+        plt.show()
+
```

Comment:
This may be a personal preference for a crutch, but I like to be able to read inline comments for each step of the code.  That way I can easily read what the code is doing and then read the code.  It seems to me doc strings do a good job of explaining variables but they dont explain the various steps along the way.  I wonder if I used inline comments?

Reply:
you did use a ton of inline comments, and I greatly appreciated it while reviewing your code. I wasn't thinking about the user's ease of reading my code while I was writing, since no one usually reads my work. Now I see the usefulness, and will include more inline comments for shared work.